### ISSUE TITLE
Support gzip

### ISSUE THREAD

**Issue by wgh136** [Tienisto/rhttp#36]
Thank you for the nice package!
It appears that the library does not support gzip.

Use following code to start a server:
```dart
import 'dart:convert';
import 'dart:io';

import 'package:shelf/shelf.dart';
import 'package:shelf/shelf_io.dart' as shelf_io;

void main() async {
  var handler = const Pipeline().addMiddleware(logRequests()).addHandler((request) {
    return Response(
      200,
      body: gzip.encode(utf8.encode('Hello, World!')),
      headers: {
        'Content-Type': 'text/plain',
        'Content-Encoding': 'gzip',
      },
    );
  });

  await shelf_io.serve(handler, 'localhost', 8080);
}
```

Use the following code to get response:
```dart
void test() {
  Rhttp.get("http://localhost:8080").then((value) {
    print(value.body);
  });
}
```
The result is
![image](https://github.com/user-attachments/assets/51cf87f1-d0ba-4b96-9ff7-85d86b914f6e)

Using dio, it works as expected:
```dart
void test() {
  Dio().get("http://localhost:8080").then((value) {
    print(value.data);
  });
}
```
The result is
![image](https://github.com/user-attachments/assets/4d465643-7826-49ef-a3d9-963784acf63d)

**Comment by Tienisto** [comment]
Thanks for noticing this bug. It seems that `gzip` needs to be enabled in `reqwest` (https://stackoverflow.com/questions/60470052/why-does-the-reqwest-http-library-return-binary-data-instead-of-a-text-body)

**Comment by Tienisto** [comment]
Fixed in 17c4b7344c79bbe4ba60db899a0595fda58cb8fb
